test/test_modified_scripts.py

```diff
@@ -0,0 +1,162 @@
+
+
+def install_modified():
+    """Installs modified scripts and returns any errors found"""
+    try:
+        import retriever as rt
+    except ImportError:
+        print("Retriever is not installed. Skipping tests...")
+        return
+
+    setup_module()
+
+    spatial_datasets = [
+        "forest-inventory-analysis",
+        "prism-climate",
+        "vertnet",
+        "NPN",
+        "mammal-super-tree"
+    ]
+    spatial_datasets_list = [dataset.lower() for dataset in spatial_datasets]
+
+    modified_scripts = get_modified_scripts()
+    if modified_scripts is None:
+        print("No new scripts found. Database is up to date.")
+        sys.exit()
+    if os.path.exists("test_modified"):
+        subprocess.call(['rm', '-r', 'test_modified'])
+    os.makedirs("test_modified")
+    os.chdir("test_modified")
+    dbfile = os.path.normpath(os.path.join(os.getcwd(), 'testdb_retriever.sqlite'))
+
+    engine_test = {
+        rt.install_xml: ["xml", {'table_name': '{db}_{table}.xml', 'debug': True}],
+
+        rt.install_json: ["json", {'table_name': '{db}_{table}.json', 'debug': True}],
+
+        rt.install_csv: ["csv", {'table_name': '{db}_{table}.csv', 'debug': True}],
+
+        rt.install_sqlite: ["sqlite", {'file': dbfile,
+                            'table_name': '{db}_{table}', 'debug': True}]
+    }
+    
+    errors = []
+    for script in modified_scripts:
+        if script in spatial_datasets_list:
+            args = {'user': 'postgres',
+                    'password': os_password,
+                    'host': pgdb_host,
+                    'port': 5432,
+                    'database': testdb_retriever,
+                    'database_name': 'testschema',
+                    'table_name': '{db}.{table}',
+                    'debug': True}
+            cmd = 'psql -U postgres -d ' + testdb_retriever +' -h ' + pgdb_host + ' -w -c \"DROP SCHEMA IF EXISTS testschema CASCADE\"'
+            subprocess.call(shlex.split(cmd))
+            try:
+                rt.install_postgres(script, **args)
+            except KeyboardInterrupt:
+                pass
+            except Exception as e:
+                print("ERROR.")
+                errors.append(("postgres", script, e))
+            continue
+        for install_function in engine_test:
+            args = engine_test[install_function]
+            try:
+                install_function(script, **(args[1]))
+            except KeyboardInterrupt:
+                pass
+            except Exception as e:
+                print("ERROR.")
+                errors.append((args[0], script, e))
+
+    os.chdir("..")
+    subprocess.call(['rm', '-r', 'test_modified'])
+
+    if errors:
+        print("Engine, Dataset, Error")
+        for error in errors:
+            print(error)
+        exit(1)
+    else:
+        print("All tests passed. All scripts are updated to latest version.")
+        exit(0)
+
+if __name__ == "__main__":
+    install_modified()
```

Comment:
This is the same as the `test_modified_scripts.py` present in `retriever`, where it attempts to install the modified dataset scripts in various engines.